src/Synchronizer.ts

```
@@ -106,14 +107,26 @@ export class Synchronizer {
       return existing;
     }
 
-    if (undefined === knownUrn) {
-      // C++ calls GetParams().QueryDocumentURN()
-    }
+    // TODO: This if statement does nothing.
+
+    // if (undefined === knownUrn) {
+    //   // C++ calls GetParams().QueryDocumentURN()
+    // }
+
+    // TODO: Create document with scope pointing to the new RepositoryLink? Currently the scope
+    // must be known before creating the link.
+
+    // By default, assume that the identifier of each document is unique to the repository.
+    // 'Recommended to point to a RepositoryLink when ids are unique per repository.'
+    // https://www.itwinjs.org/bis/domains/biscore.ecschema/#externalsourceaspect
+
     const repositoryLink = this.makeRepositoryLink(sourceItem.id, "", knownUrn);
 
-    if (undefined === repositoryLink) {
-      throw new IModelError(IModelStatus.BadElement, `Failed to create repositoryLink for ${knownUrn}`);
-    }
+    // TODO: makeRepositoryLink can't return undefined.
```

by 'can't' do you mean it's impossible and therefore you're commenting out this test b/c it's unnecessary OR should we throw an error here?

---

`createElement` in `makeRepositoryLink` throws. The return type of `makeRepositoryLink` is a `RepositoryLink` that's been narrowed (in type) to an `Element`.

We're not using TypeScript's strict mode, that's been added in the `coverage` branch that's now merged. I think the author may have been trying to guard against `strictNullChecks` but I'm not sure.